src/algebra/gradient.jl

```diff
@@ -103,6 +103,19 @@ function Gradient(
     MapOver(grad_shape_functionsNA(fs, n, ctype, cnodes, ξ))
 end
 
+function Gradient(
+    cellFunction::AbstractCellFunction{<:PhysicalDomain, S},
+    cPoint::CellPoint,
+) where {S}
+    f = get_function(cellFunction)
+    x = change_domain(cPoint, PhysicalDomain()) # transparent if already in PhysicalDomain
+    return _gradient_or_jacobian(Val(S), f, get_coord(x))
```

Comment:
@ghislainb can you please explain me why we need a `Val` here since `S` is already a `Val` (according to `PhysicalFunction` constructor)

Reply:
Because  `PhysicalFunction` calls the `CellFunction` constructor where `S` is extracted from `Val{S}` and used as a **value-parameter**, see:  https://github.com/bcube-project/Bcube.jl/blob/fed0a38f495189050198875ecd2a16acf76d5f33/src/cellfunction/cellfunction.jl#L218-L220

Reply:
Ok but then, schematically we have `PhysicalFunction(f, 2) => CellFunction(..., Val(2)) => CellFunction{f, 2}(...)` ? Isn't it weird? I mean : the `Val` here seems useless since in the end the type parameter of `CellFunction` has no `Val`...

Reply:
No it's not weird. Actually we have:
- `PhysicalFunction(f, 2) => PhysicalFunction(f, Val(2))`  which is not type-stable but it provides an easy-to-use constructor for end-users
- `PhysicalFunction(f, Val(2)) => CellFunction(..., Val(2))` which is type-stable and recommended for internal use in Bcube
- `CellFunction(..., Val(2)) => CellFunction{f, 2}(...)`, this is the common way to do a type-stable constructor which initializes a `struct` with a `value` (here `2`) as a parameter. Note that `CellFunction(..., 2) => CellFunction{f, 2}(...)` would not be type-stable since an integer type as second argument can initialize different type of `CellFunction` depending on its value.

I hope that is clear.

Reply:
If I understand correctly, this is only to have a type-stable **constructor** of a `CellFunction` right? But is this type-stable constructor necessary for efficient gradients of PhysicalFunction? Or is it just a miscellaneous improvement?